syntax = "proto3";

package plosolver.common;

option go_package = "github.com/plosolver/backend/protos/common";
option java_multiple_files = true;
option java_package = "com.plosolver.grpc.common";
option java_outer_classname = "CommonProto";

import "google/protobuf/timestamp.proto";

// =============================================================================
// Common Messages
// =============================================================================

message User {
  int32 id = 1;
  string email = 2;
  string username = 3;
  bool is_active = 4;
  bool is_verified = 5;
  google.protobuf.Timestamp created_at = 6;
  google.protobuf.Timestamp updated_at = 7;
  string subscription_status = 8;
  int32 credits_remaining = 9;
}

message Error {
  string code = 1;
  string message = 2;
  map<string, string> details = 3;
}

message PaginationRequest {
  int32 page = 1;
  int32 per_page = 2;
  string sort_by = 3;
  string sort_order = 4;
}

message PaginationResponse {
  int32 page = 1;
  int32 per_page = 2;
  int32 total = 3;
  int32 total_pages = 4;
}
